chall05/srouhe.rb

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/ruby
+# ruby 2.5.1p57
+
+require 'json'
+require 'net/http'
+
+VISITED = []
+ARG = ARGV[0]
+
+
+def shut_down
+	puts "\nNo patience, huh?"
+	sleep 1
+end
+
+
+Signal.trap("INT") { 
+  shut_down 
+  exit
+}
+
+
+def get_first_link(res)
+	begin
+		i = res.body.index('<p>')
+		main = res.body.slice(i..-1)
+		links = main.scan(/(?<=<a href="\/wiki\/).[^"]*/)
+		links.each { |link|
+			if not link.include? ":"
+				if VISITED.include? link
```

Comment:
even if ruby allows it I highly recommend not passing parameters out of parenthesis. To keep it consistent with other languages you develop or will develop with.